source/base/qprojector.cc

```diff
@@ -428,67 +485,25 @@ QProjector<dim>::project_to_face(const ReferenceCell       &reference_cell,
   AssertIndexRange(orientation, reference_cell.n_face_orientations(face_no));
   AssertDimension(reference_cell.get_dimension(), dim);
 
-  std::vector<Point<dim>> q_points;
-  std::vector<double>     q_weights = quadrature.get_weights();
-  q_points.reserve(quadrature.size());
-  if constexpr (dim == 1)
-    {
-      AssertDimension(quadrature.size(), 1);
-      q_points.emplace_back(static_cast<double>(face_no));
-    }
-  else if constexpr (dim == 2)
-    {
-      if (reference_cell == ReferenceCells::Triangle)
-        // use linear polynomial to map the reference quadrature points
-        // correctly on faces, i.e., BarycentricPolynomials<1>(1)
-        for (unsigned int p = 0; p < quadrature.size(); ++p)
-          {
-            if (face_no == 0)
-              q_points.emplace_back(quadrature.point(p)[0], 0);
-            else if (face_no == 1)
-              q_points.emplace_back(1.0 - quadrature.point(p)[0],
-                                    quadrature.point(p)[0]);
-            else if (face_no == 2)
-              q_points.emplace_back(0, 1.0 - quadrature.point(p)[0]);
-            else
-              DEAL_II_ASSERT_UNREACHABLE();
-          }
-      else if (reference_cell == ReferenceCells::Quadrilateral)
-        for (unsigned int p = 0; p < quadrature.size(); ++p)
-          {
-            if (face_no == 0)
-              q_points.emplace_back(0, quadrature.point(p)[0]);
-            else if (face_no == 1)
-              q_points.emplace_back(1, quadrature.point(p)[0]);
-            else if (face_no == 2)
-              q_points.emplace_back(quadrature.point(p)[0], 0);
-            else if (face_no == 3)
-              q_points.emplace_back(quadrature.point(p)[0], 1);
-            else
-              DEAL_II_ASSERT_UNREACHABLE();
-          }
-      else
-        DEAL_II_ASSERT_UNREACHABLE();
-
-      if (orientation == numbers::reverse_line_orientation)
-        {
-          std::reverse(q_points.begin(), q_points.end());
-          std::reverse(q_weights.begin(), q_weights.end());
```

Comment:
this patch fixes another minor bug where we didn't scale the quadrature weights of the second face of a `Triangle` correctly 🤦🏻